src/posix/sdl/i_input.cpp

```diff
@@ -306,55 +306,46 @@ void MessagePump (const SDL_Event &sev)
+			*/
+			event.type = sev.type == SDL_MOUSEBUTTONDOWN ? EV_KeyDown : EV_KeyUp;
+			switch (sev.button.button)
 			{
-				event.type = sev.type == SDL_MOUSEBUTTONDOWN ? EV_KeyDown : EV_KeyUp;
-				/* These button mappings work with my Gentoo system using the
-				* evdev driver and a Logitech MX510 mouse. Whether or not they
-				* carry over to other Linux systems, I have no idea, but I sure
-				* hope so. (Though buttons 11 and 12 are kind of useless, since
-				* they also trigger buttons 4 and 5.)
-				*/
-				switch (sev.button.button)
-				{
-				case SDL_BUTTON_LEFT:	event.data1 = KEY_MOUSE1;		break;
-				case SDL_BUTTON_MIDDLE:	event.data1 = KEY_MOUSE3;		break;
-				case SDL_BUTTON_RIGHT:	event.data1 = KEY_MOUSE2;		break;
-				case 8:		event.data1 = KEY_MOUSE4;		break; // For whatever reason my side mouse buttons are here.
-				case 9:		event.data1 = KEY_MOUSE5;		break;
-				case SDL_BUTTON_X1:		event.data1 = KEY_MOUSE6;		break; // And these don't exist
-				case SDL_BUTTON_X2:		event.data1 = KEY_MOUSE7;		break;
-				case 6:		event.data1 = KEY_MOUSE8;		break;
-				default:	printf("SDL mouse button %s %d\n",
-					sev.type == SDL_MOUSEBUTTONDOWN ? "down" : "up", sev.button.button);	break;
-				}
-				if (event.data1 != 0)
-				{
-					D_PostEvent(&event);
-				}
+			case SDL_BUTTON_LEFT:	event.data1 = KEY_MOUSE1;		break;
+			case SDL_BUTTON_MIDDLE:	event.data1 = KEY_MOUSE3;		break;
+			case SDL_BUTTON_RIGHT:	event.data1 = KEY_MOUSE2;		break;
+			case SDL_BUTTON_X1:		event.data1 = KEY_MOUSE4;		break;
+			case SDL_BUTTON_X2:		event.data1 = KEY_MOUSE5;		break;
+			case 6:		event.data1 = KEY_MOUSE6;		break;
+			case 7:		event.data1 = KEY_MOUSE7;		break;
+			case 8:		event.data1 = KEY_MOUSE8;		break;
+			default:	printf("SDL mouse button %s %d\n",
+				sev.type == SDL_MOUSEBUTTONDOWN ? "down" : "up", sev.button.button);	break;
+			}
+			if (event.data1 != 0)
+			{
+				D_PostEvent(&event);
 			}
 		}
-		else if (sev.type == SDL_MOUSEMOTION || (sev.button.button >= 1 && sev.button.button <= 3))
+		else if ((sev.button.button >= SDL_BUTTON_LEFT && sev.button.button <= SDL_BUTTON_X2))
 		{
 			int x, y;
 			SDL_GetMouseState (&x, &y);
 
 			event.data1 = x;
 			event.data2 = y;
 
-			screen->ScaleCoordsFromWindow(event.data1, event.data2);
-
 			event.type = EV_GUI_Event;
-			if(sev.type == SDL_MOUSEMOTION)
-				event.subtype = EV_GUI_MouseMove;
-			else
-			{
-				event.subtype = sev.type == SDL_MOUSEBUTTONDOWN ? EV_GUI_LButtonDown : EV_GUI_LButtonUp;
-				event.subtype += (sev.button.button - 1) * 3;
-			}
+			event.subtype = sev.type == SDL_MOUSEBUTTONDOWN ? EV_GUI_LButtonDown : EV_GUI_LButtonUp;
+			if (sev.button.button == SDL_BUTTON_X1)
+				event.subtype += EV_GUI_BackButtonDown - EV_GUI_LButtonDown;
+			else if (sev.button.button == SDL_BUTTON_X2)
+				event.subtype += EV_GUI_FwdButtonDown - EV_GUI_LButtonDown;
+			else event.subtype += (sev.button.button - 1) * 3;
```

Comment:
I think this should be rewritten this with `switch` statement or something like that.
I know that Windows backend has similar code, although it's really hard to grasp what it does.

Reply:
Do you mean that the else condition should also be turned into a switch chain, or should it just have two explicit conditions with a default? The reason I added these branches is because there's a set of events between EV_GUI_RButtonDblClick and EV_GUI_BackButtonDown, so the enum integer trick wouldn't work with the side mouse buttons.

Reply:
I meant, this math looks weird, and it's really hard to comprehend. Just compare it with `switch` from `if` block. It's obvious what's going on there.

Reply:
Alright, I just pushed a change to make the code handling this really explicit (although tell me if it's too explicit now, heh).

Reply:
Yeah, it's a bit too explicit indeed. Actually, I was thinking about one `switch` with ternary operator in each `case`, but this one looks OK too.